Il2CppInterop.Common/AssemblyIATHooker.cs

```diff
@@ -0,0 +1,660 @@
+﻿using System.Runtime.InteropServices;
+
+// ReSharper disable BuiltInTypeReferenceStyle
+
+
+namespace Il2CppInterop.Common
```

Comment:
I didn't read over this file too closely, but a lot of it looks very similar to the AsmResolver code.